app/services/service_provider_request_handler.rb

```diff
@@ -34,8 +36,19 @@ def current_sp
   end
 
   def sp_stored_in_session
+    return if service_provider_request_proxy.blank?
+    service_provider_request_proxy.issuer
+  end
+
+  def sp_url_stored_in_session
+    return if service_provider_request_proxy.blank?
+    service_provider_request_proxy.url
```

Comment:
could we nil-safe navigate these?
```suggestion
    service_provider_request_proxy&.issuer
  end

  def sp_url_stored_in_session
    service_provider_request_proxy&.url
```